### DISCUSSION TITLE
Use PostFXPlugin (such as OutlinePipeline) with ContainerLite

### DISCUSSION THREAD

**Discussion by PtrBld** [rexrainbow/phaser3-rex-notes#311] [Q&A] [answered]
Hi, 

I wanted to ask if it is possible to use a post fx pipeline with a containerlite gameobject? I cannot get it to work...
If I try to do it like this:
```ts
postFxPlugin.add(gameObject, {
   thickness: 3,
    outlineColor: 0xff8a50
});
```

 I get the following error:
```
ERROR TypeError: gameObject.setPostPipeline is not a function
    at OutlinePipelinePlugin.add (BasePostFxPipelinePlugin.js:22:1)
    at OutlinePipelinePlugin.add (outlinepipeline-plugin.js:15:1)
```

However this works fine if I use a ``Phaser.GameObjects.Container``.

Thanks.

**Comment by rexrainbow** [comment, marked as answer]
containerLite does not have render logic. 
You can [put containerLite and its children into container, or layer](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/containerlite/#add-to-container), then apply shader effect on container, or layer.
If containerLite does not have mask children, it is fine to put into container.
Or, `var layer = container.getLayer()` will [return a private layer](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/containerlite/#layer) for this containerLite.

**Reply by PtrBld** [reply]
Thanks very much. Works great!
